pipeline/00-ingest.R

```diff
@@ -322,17 +323,31 @@ training_data_clean <- training_data_fil %>%
       TRUE,
       sv_is_outlier
     ),
-    sv_outlier_type = ifelse(
-      meta_modeling_group == "NONLIVABLE",
-      "Non-livable area",
-      sv_outlier_type
+    # Assign 'Non-livable area' to the first available outlier reason column
+    # as to not replace
+    sv_outlier_reason1 = case_when(
+      meta_modeling_group == "NONLIVABLE" &
+        is.na(sv_outlier_reason1) ~ "Non-livable area",
+      TRUE ~ sv_outlier_reason1
+    ),
+    sv_outlier_reason2 = case_when(
+      meta_modeling_group == "NONLIVABLE" &
+        is.na(sv_outlier_reason1) &
+        is.na(sv_outlier_reason2) ~ "Non-livable area",
+      TRUE ~ sv_outlier_reason2
+    ),
+    sv_outlier_reason3 = case_when(
+      meta_modeling_group == "NONLIVABLE" &
+        is.na(sv_outlier_reason1) &
+        is.na(sv_outlier_reason2) &
+        is.na(sv_outlier_reason3) ~ "Non-livable area",
+      TRUE ~ sv_outlier_reason3
```

Comment:
Push the "NONLIVABLE" value to the first empty outlier_reason column

Reply:
I like the approach you've outlined, but curious to see what @jeancochrane and @wrridgeway think. We'd need to make sure that the non-livable units are booted appropriated later in the pipeline i.e. any time _any_ column has a non-livable flag.

Reply:
I didn't specify this explicitly in my comment. To clarify: we can add the rolling methodology, unless there are already 3 outlier reasons, in which case we replace the third reason.

Reply:
I don't understand the point of this since I'm not privy to the motivation for this update. Sales val doesn't get applied to non-livable areas, yeah? Why do we need to note non-livable status as being a reason for outlier status then?

Reply:
If we don't care about persisting the NONLIVABLE description, then this doesn't matter. If we excluded this sale, looked in the training data, and then didn't find any reason for the exclusion it could be confusing. What do you think @dfsnow ?

Reply:
I still don't quite understand @wagnerlmichael, aren't there only three options for how a nonlivable sale gets excluded:

1. The sale was not excluded in sales val, in which case it won't have all three `sv_outlier_reason` columns and we will be able to push `Non-livable area` to one of those reasons
2. The sale _was_ excluded in sales val, but has at least one open `sv_outlier_reason` column, in which case we will be able to push `Non-livable area` to one of those reasons
3. The sale _was_ excluded in sales val, but has no open `sv_outlier_reason` columns, in which case it doesn't matter that we drop the `Non-livable area` reason because we have three other traceable reasons

Am I right, or is there another case where we might end up with an excluded sale that we can't account for?

Reply:
That's right. The "Non-livable" comes up for reference in some output generated for analysts downstream - I think.

After a discussion with Dan, we have decided to impute "Non-livable" into the `sv_outlier_reason1` as an override, and clear out all of the other outlier reason columns.